### ISSUE TITLE
Trakt import mismatching TVDB-only shows

### ISSUE THREAD

**Issue by Kernald** [sbondCo/Watcharr#646]
Some shows on Trakt don't exist on TMDB (from before their migration a few years ago) - e.g. [Formula 1](https://trakt.tv/shows/formula-1). It seems that despite that, when importing from Trakt, Watcharr tries to match views from those shows and ends up creating events in other shows which don't necessarily have the matching seasons or episodes (e.g. I have entries in Taskmaster for S2023E88).

**Comment by IRHM** [comment]
Hi @Kernald, thanks for opening this issue.

I quickly made a new Trakt account, added two Foruma 1 episodes and tried to import and looks like it shows an error for me:
![image](https://github.com/user-attachments/assets/8b0ca1f7-4e03-435e-a0e9-8b761609ed5f)

Did that not show for you?

If it did not, could you let me know what Watcharr version you are running and if you could try again with debug logging enabled, if you see anything weird going on.

**Comment by Kernald** [comment]
I haven't seen the end of the import dialog - I have a 10+ years history on Trakt so it took a long, long time. I had to reload the tab at some point, I'm not too sure when/how the dialog is supposed to show up, so I'm not sure whether not seeing it is intended or not.

I haven't deployed Watcharr myself yet, this was on the instance at beta.watcharr.app two days ago. Would it be useful for me to try again locally?

**Comment by IRHM** [comment]
@Kernald Oh okay I see, I haven't tested the importer with such a large history, that certainly sounds like it would take forever to import.

Currently, after you start the import, you'll have to leave the importing popup open until everything finishes, then it'll show the overview (there's no way to get back to it one its closed).

If you try locally, you should be able to follow the import process more closely in the logs if you tail them. When it gets to Formula 1, you should see it log `processTraktHistoryItem: Item had no tmdbId. Cannot process.`.

One thing to note with the logs though is that (especially with debug enabled), the importing will log so much that it will likely rotate through multiple log files (and probably delete a few too).

p.s. Is `Taskmaster` a movie/show? I couldn't find it searching that name.

**Comment by Kernald** [comment]
Cheers - I'll try locally when I've got a bit more time. Taskmaster is indeed a TV show: https://www.themoviedb.org/tv/63404-taskmaster - but for what it's worth, I had multiple entries in multiple TV shows with a season number being a year, matching the watch date of F1 races/episodes on TVDB - The Simpsons, House of Cards, The Office... None of these shows have season numbers in the thousands.

**Comment by IRHM** [comment]
I checked my history, I was searching Ticketmaster each time ;( i dont have brain anymore

That sounds good let me know how it goes.

By an `entry`, is it adding activity that looks like it was meant for F1 in the other shows? That sounds concerning, will have a look.

**Comment by Kernald** [comment]
> By an entry, is it adding activity that looks like it was meant for F1 in the other shows? That sounds concerning, will have a look.

Correct. If I go on e.g. Taskmaster's page, in the Activity section, I can see e.g. "S2022E94 Added as Finished" on the 23rd of October 2022. Sure enough, that's two days after [this](https://thetvdb.com/series/formula-1/episodes/9018031). Similarly, I see some seasons (e.g. 2023) marked as Watching (because some episodes in the season have been marked as finished). On the home page, in the list, Taskmaster is displayed as with S2023E88 as the next episode to watch (or is it last watched?) if I enable "Watching Season".
